test/wordsearch.js

```diff
@@ -0,0 +1,79 @@
+/* global describe, it */
+
+const assert = require('assert');
+const { WordSearchSolver } = require('../');
+
+describe('WordSearchSolver', function () {
+  it('Cardinal directions', function () {
+    const matrix = [
+      [ 'p', 'x', 'x', 'x' ],
+      [ 'u', 'w', 'i', 'n' ],
+      [ 'z', 'x', 'x', 'x' ],
+      [ 'z', 'x', 'x', 'x' ]
+    ];
+    const solver = new WordSearchSolver(matrix);
+    const results = solver.findWords(['puzz', 'win', 'foo', 'bar', 'baz']);
+
+    assert.strictEqual(results.length, 2);

```

Comment:
I wonder if it would be worth adding a couple more asserts that verify that the results array contains the exact words we're expecting.

Reply:
Generally I'd recommend asserting the length before going further, also you should be able to simply do `results.indexOf('puzz') >= 0`.  It's likely faster than iterating yourself.